proposals/ideas/xxx-flashflow.txt

```diff
@@ -0,0 +1,713 @@
+available (semi?) publicly for study.
+
+In the long term experiments will be performed to study ways of using FF v3bw
+files to improve load balancing. Two examples: (1) using FF v3bw files instead
+of sbws's (and eventually phasing out torflow/sbws), and (2) continuing to run
+sbws but use FF's results as a better estimate of relay capacity than observed
+bandwidth. Authentication and other FlashFlow features necessary to make it
+completely ready for full production deployment will be worked on during this
+long term phase.
+
+3. FlashFlow measurement system: Short term
+
+The core measurement mechanics will be implemented in little-t tor, but a
+separate codebase for the FlashFlow side of the measurement system will also
+be created. This section is divided into three parts: first a discussion of
+changes/additions that logically reside entirely within tor (essentially:
+relay-side modifications), second a discussion of the separate FlashFlow
+code that also requires some amount of tor changes (essentially: measurer-side
+and coordinator-side modifications), and third a security discussion.
+
+3.1 Little-T Tor Components
+
+The primary additions/changes that entirely reside within tor on the relay
+side:
+
+  - New torrc options/consensus parameters.
+  - New cell types.
+  - Pre-measurement handshaking (with a simplified authentication scheme).
+  - Measurement mode, during which the relay will echo traffic with measurers,
+    set a cap on the amount of background traffic it transfers, and report the
+    amount of transferred background traffic.
+
+3.1.1 Parameters
+
+FlashFlow will require some consensus parameters/torrc options. Each has some
+default value if nothing is specified; the consensus parameter overrides this
+default value; the torrc option overrides both.
+
+FFMeasurementsAllowed: A global toggle on whether or not to allow
+measurements. Even if all other settings would allow a measurement, if this is
+turned off, then no measurement is allowed. Possible values: 0, 1. Default: 0
+(disallowed).
+
+FFAllowedCoordinators: The list of coordinator TLS certificate fingerprints that are
+allowed to start measurements. Relays check their torrc when they receive a
+connection from a FlashFlow coordinator to see if it's on the list. If they
+have no list, they check the consensus parameter. If nether exist, then no
+FlashFlow deployment is allowed to measure this relay. Default: empty list.
+
+FFMeasurementPeriod: A relay should expect on average, to be measured by each
+FlashFlow deployment once each measurement period. A relay will not allow
+itself to be measured more than twice by a FlashFlow deployment in any time
+window of this length. Relays should not change this option unless they really
+know what they're doing. Changing it at the relay will not change how often
+FlashFlow will attempt to measure the relay. Possible values are in the range
+[1 hour, 1 month] inclusive. Default: 1 day.
+
```

Comment:
Does this allow one measurement per period per coordinator?

Reply:
Yes. I'm using "FlashFlow deployment" as a different (better?) way of saying "a FlashFlow coordinator".